warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyIndexIterator.java

```diff
@@ -194,6 +205,21 @@ public void next() throws IOException {
                 continue;
             }
             
+            // only inspect the values specified in the range since a broad row or uid range will potentially go cross document
+            if (scanRange.isStartKeyInclusive() && key.getFieldValue().compareTo(startKeyParser.getFieldValue()) < 0) {
```

Comment:
Do we need to check the field name or are we guaranteed the range is reduced to those of interest?

Reply:
line 176 checks that the cq ends with the field name, technically we should test \0<fieldname> for 100% confidence

Reply:
I believe the IteratorBuildingVisitor which ultimately creates the range will prevent creating a range outside the target field, but we can update the TermFrequencyIndexIterator to protect against it anyway